### ISSUE TITLE
Unmet Cordova dependency crashes when output isn't "pretty"

### ISSUE THREAD

**Issue by abernix** [meteor/meteor#7882]
This bug goes back to circa Meteor 1.0.2 2b1a33e903e19da551d58a7a46bcf5a490ee817b

tl;dr: If I'm not using pretty output (i.e. `METEOR_PRETTY_OUTPUT=0`), and I don't have Cordova installed, it shouldn't crash.

#### Create Project

```sh
~/Dev/Misc/meteor-tests $ meteor create no-pretty-cordova-fail && !$
Created a new Meteor app in 'no-pretty-cordova-fail'.
...snip...
```

#### Hide Android's existance (and destroy your current shell)
(or just don't have Android installed)

```sh
unset ANDROID_HOME
export PATH="$HOME/.meteor:/usr/bin:/bin:/usr/sbin:/sbin"
```

#### Try adding Android
```sh
~/Dev/Misc/meteor-tests/no-pretty-cordova-fail $ meteor add-platform android
android: added platform

Your system does not yet seem to fulfill all requirements to build apps for Android.

Please follow the installation instructions in the mobile guide:
http://guide.meteor.com/mobile.html#installing-prerequisites

Status of the individual requirements:
✓ Java JDK
✗ Android SDK: Failed to find 'ANDROID_HOME' environment variable. Try setting setting it manually.
  Failed to find 'android' command in your 'PATH'. Try update your 'PATH' to include path to valid SDK directory.
✗ Android target: Android SDK not found. Make sure that it is installed. If it is not at the default location, set the ANDROID_HOME environment variable.
✗ Gradle: Could not find gradle wrapper within Android SDK. Could not find Android SDK directory.
  Might need to install Android SDK or set up 'ANDROID_HOME' env variable.
```

:+1: 🌆 Pretty message!  Cool! ✅✅✅ 

#### Remove Android so we can try again

```sh
 ~/Dev/Misc/meteor-tests/no-pretty-cordova-fail $ meteor remove-platform android
android: removed platform
```

#### Run it again and ask for disgusting output

```sh
 ~/Dev/Misc/meteor-tests/no-pretty-cordova-fail $ METEOR_PRETTY_OUTPUT=0 meteor add-platform android
android: added platform

Your system does not yet seem to fulfill all requirements to build apps for Android.

Please follow the installation instructions in the mobile guide:
http://guide.meteor.com/mobile.html#installing-prerequisites

Status of the individual requirements:
Java JDK
/Users/jesse/.meteor/packages/meteor-tool/.1.4.1_2.1jsetbq++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/meteor-promise/promise_server.js:165
      throw error;
      ^

TypeError: printFn is not a function
    at Console._fail (/tools/console/console.js:961:14)
    at Console.failInfo (/tools/console/console.js:946:17)
    at CordovaProject.checkPlatformRequirements (/tools/cordova/project.js:321:21)
    at /tools/cli/commands-cordova.js:72:24
    at /tools/utils/buildmessage.js:271:13
    at [object Object].withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:264:29
    at [object Object].withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:262:18
    at [object Object].withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:253:23
    at [object Object].withValue (/tools/utils/fiber-helpers.js:89:14)
    at Object.capture (/tools/utils/buildmessage.js:252:19)
    at Object.main.captureAndExit (/tools/cli/main.js:271:29)
    at Command.func (/tools/cli/commands-cordova.js:42:8)
    at /tools/cli/main.js:1410:23
~/Dev/Misc/meteor-tests/no-pretty-cordova-fail $
```

👎 🔥 🚒  Ugh, it's truly horrid!

### Why?

Nope, `printFn` is not a function as you can see [in this code](https://github.com/meteor/meteor/blob/2b1a33e903e19da551d58a7a46bcf5a490ee817b/tools/console/console.js#L961).

I guess not many users are trying to use Cordova for the first time with `METEOR_NO_PRETTY` enabled and without Cordova installed.

I ran into this when running Meteor `self-test` / `ci.sh` on my own CI server.

### Ok?

Yeah, PR incoming.
